### ISSUE TITLE
Plugin breaks IntelliJ  2024.3

### ISSUE THREAD

**Issue by tomtrapp** [epam/sap-commerce-intellij-idea-plugin#1278]
I have a SAP Commerce project, after the latest IntelliJ and Plugin updates,IntelliJ is not usable anymore. It laggs, searches are not finding results, I cannot find Files by the Opening dialog, Context menu is not coming up, build errors in Java classes are not regonginzed.
I've already tried the complete "Repair IDE" path and invalidated all caches, indexes etc.

My IntelliJ is always "Re-Building typesystem"
When disabling the plugin, everything works as normal.

IntelliJ Version:
Build #IU-243.21565.193

Plugin Version:
2024.3.1

**Comment by mlytvyn** [comment]
@tomtrapp , can you try to re-import project from the scratch? haven't seen such behaviour.

**Comment by tomtrapp** [comment]
@mlytvyn 
already tried that, still the same issue

**Comment by cah-mykhailo-lytvyn** [comment]
@tomtrapp , what do you think of short sync-up in the project Slack?

**Comment by tomtrapp** [comment]
Downgrading IntelliJ worked, not everything runs again like a charm.
I'll check again next week and let you know with the latest version again, if the issue is still present, lets make a sync call

**Comment by trilokkatni** [comment]
I also have the same issue as described here with IntelliJ IDEA 2024.3.

**Comment by mlytvyn** [comment]
@trilokkatni , btw, are you using Windows/Linux/WSL/MacOS?

**Comment by trilokkatni** [comment]
I'm using MacOS Version 14.7.1

**Comment by tomtrapp** [comment]
MacOS 15.1

**Comment by mlytvyn** [comment]
very strange, nothing special, single "new" feature is `Search Everywhere`.
I've just tried and it works fine, same for few other colleagues.

it would be ideal to sync-up and try to debug it together.

**Comment by trilokkatni** [comment]
I removed the project and reimported it again(using a new name and disabling the sonarLint plugin). It did the trick for me( restart two times). Now my search is fine and no longer stuck. Indexing for itemx.xml and other background tasks were completed. Hope this helps. note* For new project import, I directly used Hybris path(not directly bin which previously I used )

**Comment by trilokkatni** [comment]
Sorry, guys! I thought the issue had been resolved and arrived early. However, whenever I open beans.xml or search for SAP types, the background indexing process starts but ends up getting stuck.

**Comment by byt3m4st3r** [comment]
**Initial condition:** same Project, all old idea config files removed, fresh built with Java version 17.0.13-sapmchn

IntelliJ 2024.2.4 and SAP Commerce IntelliJ Plugin 2024.2.4:
- IDE build ✅ 
- Unit tests ✅ 

IntelliJ 2024.3 and SAP Commerce IntelliJ Plugin 2024.3.1:
- IDE build ❌ 
- Unit tests ❌ 

Additionally:
both plugin versions create the following strange folder structure under root of the project:
```
.
── ${HYBRIS_DATA_DIR}
│   └── hsqldb
│       ├── mydb.properties
│       └── mydb.script
```

Handled in new issue: #1280

**Comment by cah-mykhailo-lytvyn** [comment]
@byt3m4st3r , it looks like another issue, can you please report it separately?

**Comment by mlytvyn** [comment]
@byt3m4st3r , it looks like another issue, can you please report it separately?

**Comment by KateZorina** [comment]
Same issue for me. At first, idea got stuck at "Re-building type system", but when it finished, search stopped working.

Intellij IDEA version:
Build #IU-243.21565.193

Plugin version:
2024.3.1

**Comment by mlytvyn** [comment]
Hi @everyone, unfortunately I'm not able to reproduce this issue and may need your help.
As for now, only we, as community, can fix it and I would like to ask for contribution from your side - grab the plugin and try to figure out what exactly does not work well for you. You can always reach me out in Slack for a short call or support.

p.s. thank you for understanding and, please, take into account, that after 2-3 years and almost solely contribution I'm not always can support all variations of the projects I have no access to.

**Comment by timoschroeder1213** [comment]
Maybe this IntelliJ issue is the cause: [Searching, Find in Files, and code highlighting/navigation are slow or broken in 2024.3](https://youtrack.jetbrains.com/issue/IJPL-171550)

**Comment by sholovii** [comment]
I was affected by the same issue, but it looks like it is solved by IntelliJ 2023.3.1 update

**Comment by cah-mykhailo-lytvyn** [comment]
> I was affected by the same issue, but it looks like it is solved by IntelliJ 2023.3.1 update

@sholovii , thank you very much! I'll close the ticket as non-plugin issue.

**Comment by mlytvyn** [comment]
> I was affected by the same issue, but it looks like it is solved by IntelliJ 2023.3.1 update

@sholovii , thank you very much! I'll close the ticket as non-plugin issue.
